### PR TITLE
test: add funcional test to textinput component

### PR DESCRIPTION
**Pull request by jenifersantoszup**
### Related Issues
#1163 

### Description and Example

Functional tests were created for the textinput component to verify that what the user types is what is being shown on the screen.

### Checklist

Please, check if these important points are met using `[x]`:

- [x] I read the [PR Guide] and followed the process outlined there for submitting this PR.
- [x] I avoided _breaking changes_ by not changing public APIs that people rely on. 
- [x] I am willing to follow-up on review comments in a timely manner.

[PR Guide]: https://github.com/ZupIT/beagle/blob/master/doc/contributing/pull_requests.md

### CODE REVIEW

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/builders/TextInputScreenBuilder.kt [hunk 1]
```diff
@@ -147,6 +152,91 @@ object TextInputScreenBuilder {
...
+                        textInputType = TextInputType.TEXT)
+                ),
+                children = listOf(
+                    TextInput(placeholder = expressionOf("@{isTextExpression.placeholder}"),
+                        type = expressionOf("@{isTextExpression.textInputType}"))
+                )
+            )
+        )
+    )
+
```
Comment (paulomeurerzup) [inline backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/builders/TextInputScreenBuilder.kt:239]: Maybe you could reduce all these functions to one parametrized function.

**Review by carlossteinzup — DISMISSED** [review]

**Review by carlossteinzup — DISMISSED** [review]

**Review by paulomeurerzup — APPROVED** [review]

File: iOS/Tests/AutomatedTestsUITests/Features/TextInput.feature [hunk 2]
```diff
@@ -27,64 +27,108 @@ Feature: TextInput Validation
...
-        | is a textInput in second plan                 |
-        | is a textInput in second plan with expression |
+            |            textInputSecondPlan                |
+            | is a textInput in second plan                 |
+            | is a textInput in second plan with expression |
 
-    Scenario Outline: TextInput 06 - validate textInput of type number
+    Scenario Outline: TextInput 06 - validate date entry when using "date" type
+        When I click in the textInput with the placeholder "<textInputTypeDate>"
+        Then validate if the text input component "<textInputTypeDate>" with the date type
```
Comment (igorgiumellizup) [inline iOS/Tests/AutomatedTestsUITests/Features/TextInput.feature:69]: These 'Then' steps seem incomplete. I'd suggest something like this:
```validate if the text input component "<textInputTypeDate>" displays date type```
```validate if the text input component "<textInputTypeEmail>" displays e-mail type```
```validate if the text input component "<textInputTypePassword>" displays password type```
and so on...

File: android/automated-tests/src/androidTest/kotlin/br/com/zup/beagle/automatedTests/cucumber/steps/TextInputSteps.kt [hunk 3]
```diff
@@ -92,10 +92,58 @@ class TextInputScreen {
...
-            .hintInSecondPlan(string)
+
+    }
+
+    @Then("^validate that the value of the text input component (.*) of type \"data\" is shown correctly$")
+    fun checkDateWriting(string: String) {
+        ScreenRobot()
+            .checkViewContainsHint(string)
+            .clickOnInputWithHint(string)
+            .typeText(string,"22/04/1500")
```
Comment (FredericoNoce) [inline android/automated-tests/src/androidTest/kotlin/br/com/zup/beagle/automatedTests/cucumber/steps/TextInputSteps.kt:103]: what does this method do (typeText)? I believe it must be more intuitive ...

File: android/automated-tests/src/androidTest/assets/features/TextInput.feature [hunk 4]
```diff
@@ -66,35 +66,75 @@ Feature: TextInput Validation
             | is a textInput in second plan                 |
             | is a textInput in second plan with expression |
 
-    Scenario Outline: TextInput 06 - validate textInput of type number
-        Then validate that a textInput <textInputTypeNumber> of type number is set
+    Scenario Outline: TextInput 06 - validate date entry when using "date" type
+        Then validate that the value of the text input component <textInputTypeDate> of type "data" is shown correctly
+
+        Examples:
+            | textInputTypeDate            |
```
Comment (FredericoNoce) [inline android/automated-tests/src/androidTest/assets/features/TextInput.feature:73]: Shouldn't this parameter be of type InputType(InputType.TYPE_CLASS_DATETIME) or TextInputType(TextInputType.DATE)?

Comment (jenifersantoszup) [reply android/automated-tests/src/androidTest/assets/features/TextInput.feature:73]: In this case, only TextInputType.DATE is used to reference one of the five types of textInput used by Beagle, ie the enums (DATE, TEXT, EMAIL, PASSWORD or NUMBER), as the focus of the test is the verification of the writing and not the type of keyboard.

**Review by hthemir — APPROVED** [review]

**Review by carlossteinzup — APPROVED** [review]

**Review by igorgiumellizup — DISMISSED** [review]

**Review by lucasfonsecazup — DISMISSED** [review]

**Review by viniciusguardieirozup — APPROVED** [review]

**Review by FredericoNoce — APPROVED** [review]

**Review by carlossteinzup — DISMISSED** [review]

File: android/automated-tests/src/androidTest/kotlin/br/com/zup/beagle/automatedTests/cucumber/robots/ScreenRobot.kt [hunk 5]
```diff
@@ -33,13 +33,7 @@ import androidx.test.espresso.assertion.ViewAssertions.matches
...
 import androidx.test.espresso.matcher.BoundedMatcher
 import androidx.test.espresso.matcher.ViewMatchers
-import androidx.test.espresso.matcher.ViewMatchers.isDisplayed
-import androidx.test.espresso.matcher.ViewMatchers.isEnabled
-import androidx.test.espresso.matcher.ViewMatchers.withClassName
-import androidx.test.espresso.matcher.ViewMatchers.withHint
-import androidx.test.espresso.matcher.ViewMatchers.withId
-import androidx.test.espresso.matcher.ViewMatchers.withInputType
-import androidx.test.espresso.matcher.ViewMatchers.withText
+import androidx.test.espresso.matcher.ViewMatchers.*
```
Comment (luisoliveirazup) [inline android/automated-tests/src/androidTest/kotlin/br/com/zup/beagle/automatedTests/cucumber/robots/ScreenRobot.kt:36]: Some imports are using the wildcard notation (*). It is best to remove them before you merge.

**Review by luisoliveirazup — APPROVED** [review]

**Review by carlossteinzup — APPROVED** [review]

**Review by carlossteinzup — APPROVED** [review]

**Review by luisoliveirazup — APPROVED** [review]

**Review by gabcoelho — DISMISSED** [review]

**Review by caioortu — DISMISSED** [review]

**Review by igorgiumellizup — APPROVED** [review]

**Review by igorgiumellizup — APPROVED** [review]

**Review by luisgustavozup — APPROVED** [review]

File: iOS/Tests/AutomatedTestsUITests/Steps/TextInputSteps.swift [hunk 6]
```diff
@@ -90,17 +90,85 @@ class TextInputSteps: CucumberStepsDefinition {
...
+        }
+
+        // Scenario 11
         Then(#"^validate textInput component of type number with text "([^\"]*)"$"#) { args, _ -> Void in
             let placeholder = args![0]
             let textField = self.application.textFields[placeholder: placeholder]
 
             XCTAssertTrue(textField?.exists ?? false)
             XCTAssertEqual(self.application.keyboards.count, 1)
+//            self.application.keys["a"].tap()
```
Comment (luismaximozup) [inline iOS/Tests/AutomatedTestsUITests/Steps/TextInputSteps.swift:168]: Why the comment?

**Review by carlossteinzup — APPROVED** [review]

**Review by luisoliveirazup — APPROVED** [review]

**Review by luismaximozup — APPROVED** [review]

**Review by luisgustavozup — APPROVED** [review]

**Review by uziasferreirazup — APPROVED** [review]

**Pull request merged**